docker-compose.yaml

```diff
@@ -73,6 +73,8 @@ services:
       - LOG_LEVEL=${LOG_LEVEL}
       - NEW_SCHEDULER=${NEW_SCHEDULER}
       - SECRET_PERSISTENCE=${SECRET_PERSISTENCE}
+      - SECRET_STORE_GCP_PROJECT_ID=${SECRET_STORE_GCP_PROJECT_ID}
+      - SECRET_STORE_GCP_CREDENTIALS=${SECRET_STORE_GCP_CREDENTIALS}
```

Comment:
Are these necessary? I don't think the scheduler should need to interact with the secrets store since it shouldn't try to rehydrate configs with secrets.

Reply:
I've to admit I just put these variables where the `SECRET_PERSISTENCE` variable was already defined without knowing the internal of the secret store logic. I'll remove this then.

Reply:
I guess the `JsonSecretsProcessor` is used in here today. [391cec4](https://github.com/airbytehq/airbyte/pull/11852/commits/391cec439df66fb6f0599de553f148965fd30074) probably should be reverted in that case since there is already a dependency that should not exist.

Longer term followup is more for @benmoriceau to think about. There's a problem with the interface if we are accessing secrets-using methods in the scheduler code. We should be able to expose non-secret config persistences to components that do not require secrets without the secrets persistence-related dependencies.

Reply:
I don't think that we will need it in the `JsonSecretsProcessor` it should be limited in the `SecretPersistence` classes which are currently used in the WorkerApp and ServerApp.